Data/Aeson/Types/FromJSON.hs

```diff
@@ -178,11 +178,11 @@ parseRealFloat _        Null       = pure (0/0)
 parseRealFloat expected v          = typeMismatch expected v
 {-# INLINE parseRealFloat #-}
 
-truncateIntegral :: Integral a => String -> Scientific -> Parser a
-truncateIntegral expected s
-    | Scientific.isInteger s = pure (truncate s)
-    | otherwise               =
-        fail $ "Floating number specified for " ++ expected ++ ": " ++ show s
+truncateIntegral :: forall a. Integral a => String -> Scientific -> Parser a
```

Comment:
Thanks for the changes! Now, I would just rename this function to something else since it's not really truncating anymore.